kernel/src/actions/mod.rs

```diff
@@ -143,8 +143,8 @@ impl Default for Format {
     }
 }
 
-#[derive(Debug, Default, Clone, PartialEq, Eq, ToSchema)]
-#[cfg_attr(test, derive(Serialize), serde(rename_all = "camelCase"))]
+#[derive(Debug, Default, Clone, PartialEq, Eq, ToSchema, Serialize, Deserialize)]
```

Comment:
This does increase compile time and binary size. What's the use case, and maybe we could feature flag it that we only derive those if `internal_api` is set?

Reply:
Makes sense, it feature flagged now.

Use-case-wise: To see the datafusion engine in action, I build a delta sharing server using it. There we need to return protocol / metadata to the client and parse it there.